### PR TITLE
[fix][api]:Fix the issue of unset total page in pagination

### LINKED ISSUE #15606: [Bug] Version 3.2.1 [Instance list paging is incorrect, the monitored disk capacity is displayed as 0]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

刚升级到3.2.1，发现实例列表分页失败，还有监控的磁盘容量显示错误 全是0 
bug1
bug2
bug3

### What you expected to happen

解决实例列表无法分页问题

### How to reproduce

如上图，3.2.1版本

### Anything else

_No response_

### Version

3.2.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by JohnZp**
## Purpose of the pull request

fix #15606 about pagination
I would like to write about the issue of disk usage, but I am not sure if it can be met on different systems
```java
    private static final File BASE_DIRECTORY = new File(".");
    public static long getDiskTotal() {
        return BASE_DIRECTORY.getTotalSpace();
    }

    public static long getDiskUsed() {
        return BASE_DIRECTORY.getUsableSpace();
    }
```

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java [hunk 1]
```diff
@@ -65,10 +65,17 @@ public PageInfo(Integer currentPage, Integer pageSize) {
...
     }
 
+    public void setPaginationInfo(Integer total, List<T> totalList) {
+        setTotal(total);
+        setTotalList(totalList);
+        if (total != null && total > 0 && this.pageSize > 0) {
+            this.totalPage = (int) Math.ceil((double) total / this.pageSize);
+        }
+    }
+
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java:75]: Can you elaborate on why you are doing this?

Comment (JohnZp) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java:75]: > Can you elaborate on why you are doing this?

Cannot paginate, as described in issue #15606

**Comment by Gallardot** [comment]
Oops. I just found out that someone was already working on it, and I created a PR #15619

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/PageInfo.java:75]: I found there is a more elegent way in #15619. I would prefer merge it. @JohnZp

**Pull request closed without merging**